HHS/Head-Start-TTADP
# Slow filter fix

## Description of change

Speed up filters by changing how SQL queries are built. Instead of associated tables having the column to search aggregated together the filters now don't do any aggregating but look for individual records that match the search strings. This speeds up the filter sub-queries dramatically.

Also fixed the CSS on the filter datepicker, the bottom border is now properly shown.

**Note** I don't think the fake data we have includes program specialists attached to grants, so that filter will always return no activity reports.

## How to test

1) Go to [dev](https://tta-smarthub-dev.app.cloud.gov/activity-reports). Enter a filter for author, collaborators, grantee, or role. A good one to start with is `grantee`s that have the name "city". Note how long the initial query takes along with how long it takes to page through the table.
2) Run the same query on [staging](https://tta-smarthub-staging.app.cloud.gov/activity-reports). You should notice a considerable speed up. (dev should be much faster then staging)

Download all reports should be faster as well (when applying a filter).

I've included performance numbers in the linked ticket so we can compare once this code has been deployed.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-340

## Checklists

### Every PR



### Production Deploy


### After merge/deploy
